spec/integration/application/resource_spec.rb

```diff
@@ -36,8 +36,10 @@
       expect {
         resource.run
       }.to exit_with(0).and output(/test_resource_spec/).to_stdout
+      begin
```

Comment:
ah you'll want to wrap lines 31-38 inside the begin-ensure, so that if the test fails and raises, we ensure the type is removed like:

```ruby
it 'lists ...' do
  begin
    modulepath = ...
    ...
  ensure
    Puppet::Type.rmtype...
  end
```

Reply:
Ahh, gotcha!